### ISSUE TITLE
network_manager.py is missing attribute _requestIdToResponseWillBeSent

### ISSUE THREAD

**Issue by tiany4** [pyppeteer/pyppeteer#100]
As the title says. Setting `await page.setRequestInterception(True)` and crawling any page will cause an exception `AttributeError: 'NetworkManager' object has no attribute '_requestIdToResponseWillBeSent'`. Adding the attribute on top with `self._requestIdToResponseWillBeSent = {}` solves the issue.

**Comment by Mattwmaster58** [comment]
Yep, this is a known issue that has been fixed in the ~~`pup2.1.1`~~ `pup2.1.1_test_requestinterception`. If you need it working now, you could subclass NetworkManager and set that property, but that could lead to other things failing silently. Otherwise, you could try installing from the pup2.1.1 branch: `pip install git+https://github.com/pyppeteer/pyppeteer2@pup2.1.1_test_requestinterception`, but I would at least try wait until that branch is merged into `pup2.1.1`, watch #93 

We're working hard to bring this codebase up to date, but it's a slow process.

**Comment by tiany4** [comment]
Hey thanks for the reply. I forgot to mention that I am on the `pup2.1.1` branch, but I guess you got that covered too.

**Comment by Mattwmaster58** [comment]
Will add fixed-in-2.1.1 once that PR lands

**Comment by tiany4** [comment]
This might be the wrong place to ask, but I'm running this currently:
```
async def processRequest(interceptedRequest):
    print(interceptedRequest.resourceType)

page.on('request', lambda interceptedRequest:
            asyncio.ensure_future(processRequest(interceptedRequest)))
```
`interceptedRequest.resourceType` is always `None`.  And if I run `interceptedRequest.isNavigationRequest()`, I get `TypeError: 'bool' object is not callable`. Any idea why?

**Comment by Mattwmaster58** [comment]
> Any idea why?

Chances are I screwed up the implementation of `Request`. 

The right place to ask would be the Matrix channel if you think it's a bug. If you think your code is incorrect, then Stackoverflow or something like the [python discord](https://pythondiscord.com/) would be more appropriate I think.

**Comment by Mattwmaster58** [comment]
Forgot to mention this: There's no need to wrap your async callback in `asyncio.ensure_future` as `AsycnIOEventEmitter` automatically does that for you, ie:

```py
async def processRequest(interceptedRequest):
    print(interceptedRequest.resourceType)

page.on('request', processRequest)
```

**Comment by Mattwmaster58** [comment]
@tiany4  In case you are interested, I just marked #93 ready for review with all tests passing, which means it should be stable enough to test if you are desperate for a fix.

**Comment by tiany4** [comment]
Thanks for the notice. I think there is something wrong but that could be a bug on my end. I'm still new to python so there's that.
